tests/run_tests.py

```diff
@@ -485,149 +471,132 @@ def run_science(config_file, vic_exe, test_data_dir, out_dir):
 
-        # ------------------------------------------------------------ #
+        # Get optional kwargs for run executable
+        run_kwargs = pop_run_kwargs(test_dict)
+
         # run VIC
         test_complete = False
         test_passed = False
         test_comment = ''
         error_message = ''
 
         try:
-            returncode = vic_exe.run(test_global_file, logdir=dirs['logs'])
+            # Run the VIC simulation
+            returncode = vic_exe.run(test_global_file, logdir=dirs['logs'],
+                                     **run_kwargs)
             test_complete = True
 
-            if returncode != 0:
-                raise VICReturnCodeError('VIC return code not equal to zero, '
-                                         'check VIC logs for test')
+            # Check return code
+            check_returncode(returncode)
 
-            # -------------------------------------------------------- #
-            # check output files
+            # check output files (different tests depending on driver)
             if test_dict['check']:
-                if 'complete' in test_dict['check']:
-                    start = None
-                    end = None
-
-                for fname in glob.glob(os.path.join(dirs['results'], '*')):
-                    df = read_vic_ascii(fname, header=True)
-                    # do numeric tests
-
-                    # check that each dataframe includes all timestamps
-                    if 'complete' in test_dict['check']:
-                        if (start is not None) and (end is not None):
-                            check_completed(df, start, end)
-                        else:
-                            start = df.index[0]
-                            end = df.index[-1]
-
-                    # check for nans in the df
-                    if 'nonans' in test_dict['check']:
-                        check_for_nans(df)
-            # -------------------------------------------------------- #
-
-            # -------------------------------------------------------- #
+                fnames = glob.glob(os.path.join(dirs['results'], '*'))
+
+                # Check that the simulation completed for all grid cells
+                if 'complete' in test_dict['check'] and driver == 'classic':
+                    test_classic_driver_all_complete(fnames)
+
+                # check for nans in all example files
+                if 'output_file_nans' in test_dict['check']:
+                    if driver == 'classic':
+                        test_classic_driver_no_output_file_nans(fnames)
+                    elif driver == 'image':
+                        domain_file = os.path.join(test_data_dir,
+                                                   test_dict['domain_file'])
+                        test_image_driver_no_output_file_nans(fnames, domain_file)
+                    else:
+                        raise ValueError('unknown driver')
+
```

Comment:
no, this is raising an error if an unknown driver is specified with `output_file_nans` as a `check` option.